### PR TITLE
feat(gateway-client): improve logging upon timeout

### PR DESCRIPTION
**Pull request by CHr15F0x**
TL;DR: timeouts are not logged clearly enough, which can be misleading

--------

1. When fetching a really large block [(sepolia/99902, ~2000txns)](https://alpha-sepolia.starknet.io/feeder_gateway/get_state_update?blockNumber=99902&includeBlock=true) my local pathfinder instance just hanged. It turned out that I was getting a [decode error](https://docs.rs/reqwest/latest/src/reqwest/error.rs.html#191):
"Request failed, retrying _reason_=error decoding response body"

2. But The core reason for this error was that the reply was timing out mid-way, so there was no way decode could succeed. However the problem in the log was that we logged the `Display` impl of  `reqwest::Error` which does not take the real __source__ into account (`TimedOut`), but only the outer kind of the error (reason=decode error).

This PR fixes this behavior by enhancing the log if `error.is_timeout() == true`, so the log now looks like this:

```
Request failed, retrying. Fetching the response or parts of it timed out. Try increasing request timeout by using the `--gateway.request-timeout` CLI option. reason=error decoding response body
```

3. The real problem lies in a discrepancy in `reqwest::Error` between its [`Display` implementation](https://docs.rs/reqwest/latest/src/reqwest/error.rs.html#185) and the semantics of [`is_timeout`](https://docs.rs/reqwest/latest/src/reqwest/error.rs.html#101) as compared to `is_<other_error>`.

### CODE REVIEW

File: crates/gateway-client/Cargo.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ gateway-test-utils = { path = "../gateway-test-utils" }
 httpmock = { workspace = true }
 pathfinder-crypto = { path = "../crypto" }
 pretty_assertions_sorted = { workspace = true }
+reqwest = { workspace = true, features = ["json"] }
```
Comment (CHr15F0x) [inline crates/gateway-client/Cargo.toml:39]: Otherwise you cannot build tests only for this crate.

File: crates/pathfinder/src/bin/pathfinder/main.rs [hunk 2]
```diff
@@ -174,7 +174,12 @@ Hint: This is usually caused by exceeding the file descriptor limit of your syst
     });
     let execution_storage = storage_manager
         .create_read_only_pool(execution_storage_pool_size)
-        .context(r"")?;
+        .context(
+            r"Creating database connection pool for execution
+
+Hint: This is usually caused by exceeding the file descriptor limit of your system.
+      Try increasing the file limit to using `ulimit` or similar tooling.",
+        )?;
```
Comment (CHr15F0x) [inline crates/pathfinder/src/bin/pathfinder/main.rs:182]: I noticed this btw, I guess we want such a hint here too?

**Review by kkovaacs — APPROVED** [review]
